tdrop

```diff
@@ -268,16 +273,26 @@ update_geometry_settings_for_monitor() {
 		x_width=$(echo "$monitor_geo" | gawk -F 'x' '{print $1}')
 		y_height=$(echo "$monitor_geo" | gawk -F 'x|+' '{print $2}')
 	else
-		local wid wininfo window_x window_y monitors_info x_end y_end
-		# determine current monitor in generic way
-		wid=$(xdotool getactivewindow)
-		if [[ -z $wid ]]; then
-			# will try again after remapping or creating the dropdown
-			return 1
+		local activity_x activity_y monitors_info x_end y_end
+		if ! $pointer_screen_detection; then
+			# determine current monitor using active window
+			local wid wininfo
+			wid=$(xdotool getactivewindow)
+			if [[ -z $wid ]]; then
+				# will try again after remapping or creating the dropdown
+				return 1
+			fi
+
+			wininfo=$(xwininfo -id "$wid")
+			activity_x=$(echo "$wininfo" | gawk '/Absolute.*X/ {print $4}')
+			activity_y=$(echo "$wininfo" | gawk '/Absolute.*Y/ {print $4}')
+		else
+			# determine current monitor using pointer location
+			local pointerinfo
+			pointerinfo=$(xdotool getmouselocation --shell)
+			activity_x=$(echo "$pointerinfo" | sed -n 's/X=//p')
```

Comment:
It's not immediately clear to me what `activity` refers to. Maybe rename it to `window_or_mouse_x` or `selected_screen_x` or something more immediately obvious.